helix-core/src/syntax.rs

```diff
@@ -106,6 +126,54 @@ pub struct IndentationConfiguration {
     pub unit: String,
 }
 
+/// Configuration for auto pairs
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields, untagged)]
+pub enum AutoPairConfig {
+    /// Enables or disables auto pairing. False means disabled. True means to use the default pairs.
+    Enable(bool),
+
+    /// The mappings of pairs.
+    Pairs(HashMap<char, char>),
+}
+
+impl Default for AutoPairConfig {
+    fn default() -> Self {
+        AutoPairConfig::Enable(true)
+    }
+}
+
+impl From<&AutoPairConfig> for Option<AutoPairs> {
+    fn from(auto_pair_config: &AutoPairConfig) -> Self {
+        let auto_pairs = match auto_pair_config {
+            AutoPairConfig::Enable(false) => None,
+            AutoPairConfig::Enable(true) => Some(AutoPairs::default()),
+            AutoPairConfig::Pairs(pairs) => Some(AutoPairs::new(pairs.iter())),
+        };
+
+        debug!("auto pairs: {:#?}", auto_pairs);
```

Comment:
I actually intended to keep this here. It could be useful if users report a bug about auto pairs; we could just ask them to turn on debug logging, and that would make it easier to track down an issue related to a custom auto pair config. It should only get printed once at startup anyway. Would you really prefer this not be here?

Reply:
I'd rather print out the entire editor config inside Application::new/Editor::new. That way it's guaranteed to log once at the start of application bootup.

Reply:
Oh, that's a good idea, I'll do that. 👍